widget/completionentry.go

```diff
@@ -123,25 +130,38 @@ type navigableList struct {
 	hide            func()
 	navigating      bool
 	items           []string
+
+	customCreate func() fyne.CanvasObject
+	customUpdate func(id widget.ListItemID, object fyne.CanvasObject)
 }
 
-func newNavigableList(items []string, entry *widget.Entry, setTextFromMenu func(string), hide func()) *navigableList {
+func newNavigableList(items []string, entry *widget.Entry, setTextFromMenu func(string), hide func(),
+	create func() fyne.CanvasObject, update func(id widget.ListItemID, object fyne.CanvasObject)) *navigableList {
 	n := &navigableList{
 		entry:           entry,
 		selected:        -1,
 		setTextFromMenu: setTextFromMenu,
 		hide:            hide,
 		items:           items,
+		customCreate:    create,
+		customUpdate:    update,
 	}
 
 	n.List = widget.List{
 		Length: func() int {
 			return len(n.items)
 		},
 		CreateItem: func() fyne.CanvasObject {
+			if fn := n.customCreate; fn != nil {
```

Comment:
If at creation of n, we defaulted customCreate to create a label. There would be no need at all here for any indirection and we could just set customCreate as the type for CreateItem. Might be good to rethink the name of customCreate here in that case.

Reply:
If we did as you propose how would someone reset it later if required? Removing this check and user cannot set the value to nil or it would panic.

Reply:
I may have missed something, but n is an internal object with customCreate and customUpdate being captured by it with no way to be changed from the outside from what I understand.